src/disco/stem/fd_stem.c

```diff
@@ -0,0 +1,668 @@
+  ulong    event_seq; /* current position in housekeeping event sequence, in [0,event_cnt) */
+  ushort * event_map; /* current mapping of event_seq to event idx, event_map[ event_seq ] is next event to process */
+  ulong    async_min; /* minimum number of ticks between processing a housekeeping event, positive integer power of 2 */
+
+  /* performance metrics */
+  ulong metric_in_backp;  /* is the run loop currently backpressured by one or more of the outs, in [0,1] */
+  ulong metric_backp_cnt; /* Accumulates number of transitions of tile to backpressured between housekeeping events */
+
+  ulong metric_regime_ticks[9];    /* How many ticks the tile has spent in each regime */
+
+  if( FD_UNLIKELY( !scratch ) ) FD_LOG_ERR(( "NULL scratch" ));
+  if( FD_UNLIKELY( !fd_ulong_is_aligned( (ulong)scratch, stem_scratch_align() ) ) ) FD_LOG_ERR(( "misaligned scratch" ));
+
+  /* in_backp==1, backp_cnt==0 indicates waiting for initial credits,
+      cleared during first housekeeping if credits available */
+  metric_in_backp  = 1UL;
+  metric_backp_cnt = 0UL;
+  memset( metric_regime_ticks, 0, sizeof( metric_regime_ticks ) );
+
+  /* in frag stream init */
+
+  in_seq = 0UL; /* First in to poll */
+
+  FD_SCRATCH_ALLOC_INIT( l, scratch );
+  in = (fd_stem_tile_in_t *)FD_SCRATCH_ALLOC_APPEND( l, alignof(fd_stem_tile_in_t), in_cnt*sizeof(fd_stem_tile_in_t) );
+
+  ulong min_in_depth = (ulong)LONG_MAX;
+
+  if( FD_UNLIKELY( !!in_cnt && !in_mcache ) ) FD_LOG_ERR(( "NULL in_mcache" ));
+  if( FD_UNLIKELY( !!in_cnt && !in_fseq   ) ) FD_LOG_ERR(( "NULL in_fseq"   ));
+  if( FD_UNLIKELY( in_cnt > UINT_MAX ) )      FD_LOG_ERR(( "in_cnt too large" ));
+  for( ulong in_idx=0UL; in_idx<in_cnt; in_idx++ ) {
+
+    if( FD_UNLIKELY( !in_mcache[ in_idx ] ) ) FD_LOG_ERR(( "NULL in_mcache[%lu]", in_idx ));
+    if( FD_UNLIKELY( !in_fseq  [ in_idx ] ) ) FD_LOG_ERR(( "NULL in_fseq[%lu]",   in_idx ));
+
+    fd_stem_tile_in_t * this_in = &in[ in_idx ];
+
+    this_in->mcache = in_mcache[ in_idx ];
+    this_in->fseq   = in_fseq  [ in_idx ];
+
+    ulong depth    = fd_mcache_depth( this_in->mcache ); min_in_depth = fd_ulong_min( min_in_depth, depth );
+    if( FD_UNLIKELY( depth > UINT_MAX ) ) FD_LOG_ERR(( "in_mcache[%lu] too deep", in_idx ));
+    this_in->depth = (uint)depth;
+    this_in->idx   = (uint)in_idx;
+    this_in->seq   = 0UL;
+    this_in->mline = this_in->mcache + fd_mcache_line_idx( this_in->seq, this_in->depth );
+
+    this_in->accum[0] = 0U; this_in->accum[1] = 0U; this_in->accum[2] = 0U;
+    this_in->accum[3] = 0U; this_in->accum[4] = 0U; this_in->accum[5] = 0U;
+  }
+
+  /* out frag stream init */
+
+  cr_avail = 0UL;
+
+  out_depth  = (ulong *)FD_SCRATCH_ALLOC_APPEND( l, alignof(ulong), out_cnt*sizeof(ulong) );
+  out_seq    = (ulong *)FD_SCRATCH_ALLOC_APPEND( l, alignof(ulong), out_cnt*sizeof(ulong) );
+
+  ulong cr_min = fd_ulong_if( !out_cnt, 128UL, ULONG_MAX );
```

Comment:
Maybe call this `depth_min`

Reply:
And then keep the variable formerly called `cr_max`. It's just confusing to see a variable change names from `cr_max` to `cr_min` without changing roles